### PR TITLE
[doc] property map doc and simple airdrop example

### PR DESCRIPTION
**Pull request by areshand**
### Description
1. add a draft for working with property map off-chain
2. A simple airdrop example

### CODE REVIEW

File: developer-docs-site/docs/concepts/coin-and-token/token_airdrop/Token Airdrop Example c4d35e65f7314699a5335fb96d2d5c1f.md
Comment (davidiw) [inline developer-docs-site/docs/concepts/coin-and-token/token_airdrop/Token Airdrop Example c4d35e65f7314699a5335fb96d2d5c1f.md:14]: where is `whitelisted_addresses` coming from?

File: developer-docs-site/docs/concepts/coin-and-token/token_airdrop/Token Airdrop Example c4d35e65f7314699a5335fb96d2d5c1f.md
Comment (davidiw) [inline developer-docs-site/docs/concepts/coin-and-token/token_airdrop/Token Airdrop Example c4d35e65f7314699a5335fb96d2d5c1f.md:19]:
```suggestion
    while (i < count) {
```

File: developer-docs-site/docs/concepts/coin-and-token/Working with PropertyMap Off-Chain.md
Comment (davidiw) [inline developer-docs-site/docs/concepts/coin-and-token/Working with PropertyMap Off-Chain.md:3]: Point to the token module and maybe call it The Aptos Token standard?

Comment (davidiw) [reply developer-docs-site/docs/concepts/coin-and-token/Working with PropertyMap Off-Chain.md:3]: should we update this to take advantage of recent new primitives in Move? u16, u32, u256?

File: developer-docs-site/docs/concepts/coin-and-token/Working with PropertyMap Off-Chain.md
Comment (davidiw) [inline developer-docs-site/docs/concepts/coin-and-token/Working with PropertyMap Off-Chain.md:5]:
```suggestion
With both the value and typing, property map can be used to read and write values of heterogeneous types in a map data structure on-chain. 
```

File: developer-docs-site/docs/concepts/coin-and-token/Working with PropertyMap Off-Chain.md
Comment (davidiw) [inline developer-docs-site/docs/concepts/coin-and-token/Working with PropertyMap Off-Chain.md:9]: I think you mean the property map can be read and written to?

File: developer-docs-site/docs/concepts/coin-and-token/token_airdrop/Token Airdrop Example c4d35e65f7314699a5335fb96d2d5c1f.md
Comment (davidiw) [inline developer-docs-site/docs/concepts/coin-and-token/token_airdrop/Token Airdrop Example c4d35e65f7314699a5335fb96d2d5c1f.md:5]: we should avoid unnecessary superlatives ...

File: developer-docs-site/docs/concepts/coin-and-token/token_airdrop/Token Airdrop Example c4d35e65f7314699a5335fb96d2d5c1f.md
Comment (davidiw) [inline developer-docs-site/docs/concepts/coin-and-token/token_airdrop/Token Airdrop Example c4d35e65f7314699a5335fb96d2d5c1f.md:3]: best two define the two methods and then explain them.

Comment (davidiw) [reply developer-docs-site/docs/concepts/coin-and-token/token_airdrop/Token Airdrop Example c4d35e65f7314699a5335fb96d2d5c1f.md:3]: but you only discuss one approach in this doc

**Review by clay-aptos — APPROVED** [review]
Thanks, Bo. This LGTM with some edits I made directly, as well as some renames and moves I will conduct to the image to get it in the right location. I also suggest shortening the filenames and eliminate spaces, ex:
Working with PropertyMap Off-Chain.md >  off-chain-property-map.md

Finally, I will get this file into navigation and announce it once this PR has been merged. Thanks!

File: developer-docs-site/docs/concepts/coin-and-token/token_airdrop/Token Airdrop Example c4d35e65f7314699a5335fb96d2d5c1f.md
Comment (areshand) [reply developer-docs-site/docs/concepts/coin-and-token/token_airdrop/Token Airdrop Example c4d35e65f7314699a5335fb96d2d5c1f.md]: This is to show a basic example of doing an airdrop.  I only think of a way to use the 2-step process, which is the reason I only discuss one approach.  The other transfer mode doesn't apply here since we don't know if the whitelist address have opt-in directly transfer

File: developer-docs-site/docs/concepts/coin-and-token/token_airdrop/Token Airdrop Example c4d35e65f7314699a5335fb96d2d5c1f.md
Comment (areshand) [reply developer-docs-site/docs/concepts/coin-and-token/token_airdrop/Token Airdrop Example c4d35e65f7314699a5335fb96d2d5c1f.md]: updated

File: developer-docs-site/docs/concepts/coin-and-token/token_airdrop/Token Airdrop Example c4d35e65f7314699a5335fb96d2d5c1f.md
Comment (areshand) [reply developer-docs-site/docs/concepts/coin-and-token/token_airdrop/Token Airdrop Example c4d35e65f7314699a5335fb96d2d5c1f.md]: directly pushed

File: developer-docs-site/docs/concepts/coin-and-token/Working with PropertyMap Off-Chain.md
Comment (areshand) [reply developer-docs-site/docs/concepts/coin-and-token/Working with PropertyMap Off-Chain.md]: updated. Yes, we should. Let me create an issue in linear to track this issue and tagged you in the issue

File: developer-docs-site/docs/concepts/coin-and-token/Working with PropertyMap Off-Chain.md
Comment (areshand) [reply developer-docs-site/docs/concepts/coin-and-token/Working with PropertyMap Off-Chain.md]: updated

File: developer-docs-site/docs/concepts/coin-and-token/Working with PropertyMap Off-Chain.md
Comment (areshand) [reply developer-docs-site/docs/concepts/coin-and-token/Working with PropertyMap Off-Chain.md]: updated

File: developer-docs-site/docs/concepts/coin-and-token/token_airdrop/Token Airdrop Example c4d35e65f7314699a5335fb96d2d5c1f.md
Comment (areshand) [reply developer-docs-site/docs/concepts/coin-and-token/token_airdrop/Token Airdrop Example c4d35e65f7314699a5335fb96d2d5c1f.md]: removed

**Comment by areshand** [comment]
> Thanks, Bo. This LGTM with some edits I made directly, as well as some renames and moves I will conduct to the image to get it in the right location. I also suggest shortening the filenames and eliminate spaces, ex: Working with PropertyMap Off-Chain.md > off-chain-property-map.md
> 
> Finally, I will get this file into navigation and announce it once this PR has been merged. Thanks!

updated the names of the files

**Comment by clay-aptos** [comment]
Let's add this to the Create NFT section in the following PR after it lands:
https://github.com/aptos-labs/aptos-core/pull/6266

**Review by davidiw — COMMENTED** [review]
@clay-aptos -- are we approving and landing documents that aren't linked in any navbars?

**Comment by clay-aptos** [comment]
Hi @areshand , how can I help finish this up?

One published, I will get into site navigation and announced. I may move the image here too:
https://github.com/aptos-labs/aptos-core/tree/main/developer-docs-site/static/img

Thanks!

**Comment by clay-aptos** [comment]
> @clay-aptos -- are we approving and landing documents that aren't linked in any navbars?

Yes and intentionally! See my response in:
https://github.com/aptos-labs/aptos-core/pull/6292#issuecomment-1409019980

There is certainly precedent (before my times):
https://aptos.dev/guides/move-guides/guide-move-transactional-testing/

I have asked a few times if we can link to that doc. :-)
https://github.com/aptos-labs/aptos-core/issues/5517#issuecomment-1355450548

As I ask in issue #6292, let me know what you think of this process. I am merely trying to speed up contributions and avoid conflicts in the nav files, which have truly been churning of recent.

**Comment by areshand** [comment]
> Hi @areshand , how can I help finish this up?
> 
> One published, I will get into site navigation and announced. I may move the image here too: https://github.com/aptos-labs/aptos-core/tree/main/developer-docs-site/static/img
> 
> Thanks!

I think I need two approvals to land this PR. @davidiw let me know if you have additional comments.

**Review by lightmark — APPROVED** [review]

**Pull request merged**
